File: src/test/java/com/infotarget/codility/java/OddOccurrencesInArrayTest.java
package com.infotarget.codility.java;

import org.junit.Assert;
import org.junit.Test;

import java.util.function.ToIntBiFunction;

public class OddOccurrencesInArrayTest {

    private OddOccurrencesInArray oddOccurrencesInArray = new OddOccurrencesInArray();

    @Test
    public void testSolution1() throws Exception {
        testSolution(oddOccurrencesInArray::solution1);
    }

    @Test
    public void testSolution2() throws Exception {
        testSolution(oddOccurrencesInArray::solution2);
    }

    private void testSolution(ToIntBiFunction<int[], Integer> solution) {
        Assert.assertEquals(7, solution.applyAsInt(new int[]{9, 3, 9, 3, 9, 7, 9}, 7));
        Assert.assertEquals(0, solution.applyAsInt(new int[]{9, 3, 9, 3, 9, 0, 9}, 7));
        Assert.assertEquals(9, solution.applyAsInt(new int[]{9, 3, 9, 3, 9, 0, 9}, 1));
    }
}

File: src/test/scala/com/infotarget/codility/scala/OddOccurrencesInArrayTest.scala
package com.infotarget.codility.scala

import org.scalatest.FunSuite

class OddOccurrencesInArrayTest extends FunSuite {

  test("solution1") {
    testSolution(OddOccurrencesInArray.solution1)
  }

  test("solution2") {
    testSolution(OddOccurrencesInArray.solution2)
  }

  private def testSolution(solution: (Array[Int], Int) => Int) = {
    assert(7 === solution(Array(9, 3, 9, 3, 9, 7, 9), 7))
    assert(0 === solution(Array(9, 3, 9, 3, 9, 0, 9), 7))
    assert(9 === solution(Array(9, 3, 9, 3, 9, 0, 9), 1))
  }
}
